### DISCUSSION TITLE
fpv.py issue not taking off

### DISCUSSION THREAD

**Discussion by 4perros** [bitcraze/discussions#307] [Q&A]
hello 
Hi Kimberly thank you very much for your help.
Regarding the not taking off, yes i have a flow deck attached, i am using an older version (Rev.D) not version 2. Could this be the problem?
Regards
Rodrigo

**Comment by knmcguire** [comment]
Have you tried with another script or the flight command buttons on the cfclient to make it take off?

Could you provide more information of your problem. There are some instructions in the discussion rules for that (nr. 2) https://github.com/orgs/bitcraze/discussions/55

**Reply by 4perros** [reply]
hi
i have run other scripts, and they work, for example from the autonomous folder i tested the motion_commander_demo.py and the postiion_commander_demo.py and from the motors folder the ramp.py example and for multiranger folder the multiranger_pointcloud.py example all work fine, the fpv.py example opens a window with the camera view and it stays on the following line: Creating queue for CPXFunction.APP.
I am not sure how to test the command buttons on the cfclient. Do you mean the up and down arrows on laptop keyboard? And how can i run the example and at the same time the cfclient?
I am using Ubuntu 20.04
regards
rodrigo

**Reply by knmcguire** [reply]
With the flight command buttons I mean the 'command based flight control' panel on the Flight-control tab of the cfclient.

For the extra information, please connect the crazyflie to the cfclient with all the decks attached, open up the console tab through 'view' > 'tabs' and copy the content of the printf there.

Also, while still connected, go to 'Help' > 'About' > 'Debug'  and copy the content there as well.

**Reply by 4perros** [reply]
hi
this is the output of the console and the debug window

SYS: Crazyflie 2.0 is up and running!
SYS: Build 0:8c606d1f8f89 (2022.05) CLEAN
SYS: I am 0x313139313535511300360046 and I have 1024KB of flash!
CFGBLK: v1, verification [OK]
DECK_CORE: 2 deck(s) found
DECK_CORE: Calling INIT on driver bcFlow for deck 0
PMW: Motion chip id: 0x49:0xB6
DECK_CORE: Calling INIT on driver bcAI for deck 1
AIDECK-ROUTER: Not setting up WiFi
MPU9250 I2C connection [OK].
AK8963 I2C connection [OK].
LPS25H I2C connection [OK].
ESTIMATOR: Using Kalman (2) estimator
CONTROLLER: Using PID (1) controller
MTR-DRV: Using brushed motor driver
SYS: About to run tests in system.c.
SYS: NRF51 version: 2022.0
EEPROM: I2C connection [OK].
STORAGE: Storage check [OK].
AK8963: Self test [OK].
DECK_CORE: Deck 0 test [OK].
DECK_CORE: Deck 1 test [OK].
SYS: Self test passed!
STAB: Wait for sensor calibration...
SYS: Free heap: 13728 bytes
ESP32: I (899) SYS: Initialized
ESP32: I (899) DISCOVERY: Hostname is aideck-842988
GAP8: -- WiFi image streamer example --
GAP8: Setting up WiFi AP
GAP8: Starting camera task...
ESP32: I (3329) WIFI: wifi_init_softap finished
STAB: Ready to fly.
ESTKALMAN: State out of bounds, resetting
STAB: WARNING: stabilizer loop rate is off (904)






Cfclient
Cfclient version: 2022.9.post1+53de36d
System: linux
Python: 3.8.10
Qt: 5.15.2
PyQt: 5.15.6

Interface status
radio: Crazyradio version 0.52
UsbCdc: No information available
udp: None
prrt: No information available
cpx: None

Input readers
linuxjsdev (2 devices connected)

Input devices
None

Crazyflie
Connected: radio://0/80/2M
Firmware: 8c606d1f8f89 (CLEAN)

Decks found
bcFlow: rev=E, adr=0DEAC71E00000080
bcAI: rev=C, adr=0D078257000000AB

Sensors found
BMP388: 0
AK8963: 1
LPS25H: 1

Sensors tests
MPU6500: 1
AK8963: 1
LPS25H: 1

regards
rodrigo

**Reply by evoggy** [reply]
Hi,

Did you try the streamer example in the GAP8 repo? If so, was this streaming images? If you don't see any images in the example there could be something wrong with the GAP8 and/or ESP32 firmware, did you re-flash these to be the latest versions?

**Reply by 4perros** [reply]
hello
Yes i have tried the wifi streamer example and it works fine, i get two windows streaming images. also the fpv.py example opens a single large window with camera working as well only problem is that it just wont take off as explained in the script that as soon as you run the application it should hover at 0.3 m. I am running the script with the following command 
CFLIB_URI=tcp://192.168.4.1:5000 python3 fpv.py
regards
rodrigo

**Reply by 4perros** [reply]
hi
I tried the sbs_motion_commander.py and got this outcome:
Deck is not connected
No flow deck detected!
although i have the flow deck connected, i read through the script and found on line 114 the following:
scf.cf.param.add_update_callback(group='deck', name='bcFlow2' 
so i took the number 2 out, saved it and run it again and it worked with crazyradio connected
scf.cf.param.add_update_callback(group='deck', name='bcFlow' (edited version)
 
do you think that the fpv.py example only works with the flow deck v2, as i have an older flow deck.
regards
rodrigo

**Comment by 4perros** [comment]
hi 
yes i have a flow deck attached.

I tried the step-by-step sbs_motion_commander.py and got this outcome:
Deck is not connected
No flow deck detected!
although i have the flow deck connected, i read through the script and found on line 114 the following:
scf.cf.param.add_update_callback(group='deck', name='bcFlow2'
so i took the number 2 out, saved it and run it again and it worked with crazyradio connected
scf.cf.param.add_update_callback(group='deck', name='bcFlow' (edited version)

do you think that the fpv.py example only works with the flow deck v2, as i have an older flow deck.
regards
rodrigo

**Reply by knmcguire** [reply]
It seems that the fpv demo does not check for decks. So probably the connection is going wrong, also since the console is saying 'Not setting up WiFi'

I'm not sure what else to do you your case. You can try to go through the whole tutorial again and connect through the aideck with it being the AP.

**Reply by 4perros** [reply]
hi
i have solved the problem. I reinstalled crazyflie-python and crazyflie-firmware, then reflashed the crazyflie with the latest software, i then run the example and the crazyflie just took off. I also changed the position of wifi router and made sure bluetooth is turned off on laptop.
Is it possible to save the streaming video from this example?
than you for your help.
regards
rodrigo

**Reply by knmcguire** [reply]
Good it works now!

For the question 'Is it possible to save the streaming video from this example?' could you make a new discussion? This will be interesting for more people but asking it in a thread of another issue might make it invisible

**Reply by noahzn** [reply]
> crazyflie-python

Hi,  I meet the same problem. are you using the 2022.09 version?

**Reply by 4perros** [reply]
hi
yes i am using the 2022/09 version.
rodrigo
